control-plane/agents/src/bin/core/controller/reconciler/node/nexus.rs

```diff
@@ -109,32 +120,33 @@ async fn find_shutdown_volumes(context: &PollContext, node_id: &NodeId) -> Resul
                 draining_volumes_to_remove.insert(vi);
             }
         }
-        context
-            .specs()
-            .remove_node_draining_volumes(context.registry(), node_id, draining_volumes_to_remove)
+        guarded_node_spec
+            .remove_draining_volumes(
+                context.registry(),
+                DrainingVolumes::new(draining_volumes_to_remove),
+            )
             .await?;
     } else {
         // else the drain operation is timed out
-        context
-            .specs()
-            .remove_all_node_draining_volumes(context.registry(), node_id)
+        guarded_node_spec
+            .remove_all_draining_volumes(context.registry())
             .await?;
     }
     Ok(())
 }
 
 /// Drain the specified node if in draining state
-async fn check_and_drain_node(context: &PollContext, node_spec: &NodeSpec) -> PollResult {
-    if !node_spec.is_draining() {
+async fn check_and_drain_node(
+    context: &PollContext,
+    guarded_node_spec: &mut OperationGuardArc<NodeSpec>,
+) -> PollResult {
+    if !guarded_node_spec.as_ref().is_draining() {
         return PollResult::Ok(PollerState::Idle);
     }
-    let node_id = node_spec.id();
+    let node_id = &guarded_node_spec.as_ref().id().clone();
```

Comment:
Do we need to clone it? Can't we just get guarded_node_spec.as_ref().id when needed?

Reply:
fixed